tests/templates/test-head-management.php

```diff
@@ -0,0 +1,337 @@
+<?php
+/**
+ * Class Test_Head_Management
+ *
+ * @package WP_Irving
+ */
+
+namespace WP_Irving\Templates;
+
+use WP_Irving\Components\Component;
+use WP_UnitTestCase;
+
+/**
+ * Test head functionality.
+ *
+ * @group templates
+ */
+class Test_Head_Management extends WP_UnitTestCase {
+
+	/**
+	 * Test attachment
+	 *
+	 * @var int attachment ID.
+	 */
+	public static $attachment_id;
+
+	/**
+	 * Helper to get the attachment ID.
+	 *
+	 * @return int attachment ID.
+	 */
+	public function get_attachment_id() {
+		return self::$attachment_id;
+	}
+
+	/**
+	 * Helper to get the attachment URL.
+	 *
+	 * @return string attachment URL.
+	 */
+	public function get_attachment_url() {
+		$uploads_dir = wp_get_upload_dir();
+
+		return trailingslashit( $uploads_dir['url'] ) . 'test-image.jpg';
+	}
+
+
+	/**
+	 * Set up shared fixtures.
+	 *
+	 * @param WP_UnitTest_Factory $factory Factory instance.
+	 */
+	public static function wpSetupBeforeClass( $factory ) {
+		// Test image.
+		self::$attachment_id = $factory->attachment->create_object(
+			'test-image.jpg',
+			null,
+			[
+				'post_mime_type' => 'image/jpeg',
+				'post_type'      => 'attachment',
+				'post_excerpt'   => 'Test caption text.',
+			]
+		);
+	}
+
+	/**
+	 * Setup.
+	 */
+	public function setUp() {
+		parent::setup();
+
+		// Disable date organized uploads.
+		add_filter( 'upload_dir', [ $this, 'upload_dir_no_subdir' ] );
```

Comment:
Instead of duplicating this in several test classes, we should move this to a general helpers file that gets loaded during the bootstrapping process. Honestly, I'm unsure why the helper from the WP Unit tests wasn't working in the first place 🤷